src/detect-parse.c

```diff
@@ -2144,17 +2160,33 @@ static Signature *SigInitHelper(DetectEngineCtx *de_ctx, const char *sigstr,
     if (sig == NULL)
         goto error;
 
+    sig->sig_str = SCStrdup(sigstr);
+    if (unlikely(sig->sig_str == NULL)) {
+        goto error;
+    }
+
     /* default gid to 1 */
     sig->gid = 1;
 
-    int ret = SigParse(de_ctx, sig, sigstr, dir, &parser);
+    /* We do a first parse of the rule in a requires, or scan-only
```

Comment:
how expensive is it to this double parsing? Would it make sense to first do something like a `strstr(sigstr, "requires") != NULL` to skip most of the rules?

Reply:
may be moot if we're also parsing sid in the same pass